eos_data_distribution/ndn/file.py

```diff
@@ -264,6 +271,7 @@ def _on_complete(self):
 
     def _create_files(self, filename):
         # XXX this is racy
+        assert self._filename
```

Comment:
Whoops, yes. I also meant to remove the `self._filename` assignment from there (it fits better in the callers).